### ISSUE TITLE
issue: Input field value is not re-rendered in controlled nested array when another array element is selected 

### ISSUE THREAD

**Issue by schiller-manuel** [react-hook-form/react-hook-form#11498]
### Version Number

7.50.1

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/react-hook-form-conditional-forked-j3m6n9?file=%2Fsrc%2Findex.js%3A46%2C11

### Steps to reproduce

1. Go to [example on codesandbox](https://codesandbox.io/p/sandbox/react-hook-form-conditional-forked-j3m6n9?file=%2Fsrc%2Findex.js%3A46%2C11)
2. Switch from 0 to 1 in select
3. See that field values are not updated

see screen recording

https://github.com/react-hook-form/react-hook-form/assets/6340397/cb7714ce-10e8-4403-909d-a10750f761b5

### Expected behaviour

When the `selectedIndex` is changed from `0` to `1`, `name` changes from `test.0.values.${valueIndex}` to `test.1.values.${valueIndex}`. I would thus expect that the rendered input values change from `["aaa", "bbb", "ccc"]` to `["xxx", "yyy", "zzz"]`.

A workaround is to use the `name` as `key`, then the correct values are rendered:

```tsx
<li key={`test.${selectedIndex}.values.${valueIndex}`}>
```

https://github.com/react-hook-form/react-hook-form/assets/6340397/6f991d5c-fc85-4bca-a380-de30061459a1

### What browsers are you seeing the problem on?

Firefox, Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
could you update your CSB? it's missing.

**Comment by schiller-manuel** [comment]
sorry it was still in drafts, now it should be accessible

**Comment by bluebill1049** [comment]
the update would need to be a valid path for that registered controller input. `test.${selectedIndex}.values.${valueIndex}`, it doesn't work at your current implementation.

**Comment by schiller-manuel** [comment]
sorry I don't understand that. how would a valid path look like if mine is invalid?

**Comment by bluebill1049** [comment]
This is the input name which you have registered `test.${selectedIndex}.values.${valueIndex}`

and this is your setValue `onChange={(e) => setValue("selectedIndex", e.target.value)}`, setValue will need the specific controller name/path

**Comment by schiller-manuel** [comment]
`selectedIndex` is a field in the form state, next to `test`.

**Comment by bluebill1049** [comment]
> `selectedIndex` is a field in the form state, next to `test`.

yes, but it doesn't work like that. controller is not a state management input, it's just a input which store value for that particular path which you have giving by name/path.
